### ISSUE TITLE
Web UI cannot show default metric when training large lightgbm model

### ISSUE THREAD

**Issue by ANFANGERMI** [microsoft/nni#4882]
**Describe the issue**:
Web UI cannot show default metric when training lightgbm model with more than 700 n_estimators
![2341653310329_ pic](https://user-images.githubusercontent.com/39998643/169823411-a681a11d-d318-4d1b-b0f4-bbc2236d5f9d.jpg)
**Environment**:
- NNI version:2.6
- Training service (local|remote|pai|aml|etc):local
- Client OS:ubuntu 18.04.6 LTS
- Server OS (for remote mode only):
- Python version:3.8
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?:no
- Is running in Docker?:no

**Configuration**:
 - Experiment config (remember to remove secrets!):
 -  experiment.config.experiment_name = 'NNI_TEST_TREE'
    experiment.config.trial_concurrency = 1
    experiment.config.max_trial_number = 15
    experiment.config.search_space = search_space
    experiment.config.trial_code_directory = Path(__file__).parent
    experiment.config.tuner.name = 'GridSearch'
    experiment.config.training_service.use_active_gpu = True 
    experiment.config.debug = True
 - Search space:
 -    search_space = {
        "learning_rate": {
            "_type": "choice",
            "_value": [0.2, 0.3]
        },
        "max_depth": {
            "_type": "choice",
            "_value": [6, 7, 8]
        },
        "n_estimators":{
            "_type": "choice",
            "_value": [700, 800, 900]
        }
    }

**Log message**:
 - nnimanager.log:
 - [nnimanager.log](https://github.com/microsoft/nni/files/8754712/nnimanager.log)
 - dispatcher.log:
 - [dispatcher.log](https://github.com/microsoft/nni/files/8754707/dispatcher.log)
 - nnictl stdout and stderr:
 - launch experiment from python api
 

**How to reproduce it?**:
Train lightgbm model with n_estimators > 700 using Gridsearch tuner

**Comment by liuzhe-lz** [comment]
Have you invoked `nni.report_final_result()` in your trial code?
Maybe you can add a log message to ensure the line is executed.

**Comment by ANFANGERMI** [comment]
> Have you invoked `nni.report_final_result()` in your trial code? Maybe you can add a log message to ensure the line is executed

> Thanks for the reply!  Actually i have found that the problem is caused by a tricky usage of `nni.report_intermidiate_result()` ,instead of invoking the function in every epoch, i use a `dict` to record all the intermidiate metrics and send it to `nni.report_intermidiate_result()` all in once. Now the problem is solved by removing that line of code,but i still wonder how this happen?😄

**Comment by scarlett2018** [comment]
Closing the issue as the problem has been resolved.
